### PR TITLE
ipmikvm: add -c to use custom config file

### PR DESCRIPTION
**Pull request by quite**
This makes it possible to avoid keeping password in plaintext, and exposing on command-line, like this:

ipmikvm -c <(printf '* * someuser %s\n' "$(pass show -f servers/foo)") 10.0.0.1

### CODE REVIEW

File: ipmikvm [hunk 1]
```diff
@@ -126,22 +129,27 @@ IP=${1:-}; shift
...
-        test -z "$USER" && USER=$(echo "$LINES" | awk '{print $3}')
-        test -z "$PASS" && PASS=$(echo "$LINES" | awk '{print $4}')
-    elif test -z "$USER" && test -z "$PASS"; then
-        # No user/pass supplied. Then get all the * * matches.
-        USERS=$(awk '/^ *[^#]/{if(NF>=4&&$1=="*"&&$2=="*"){
-            printf "%s ", $3}}' "$DICT")
-        PASSES=$(awk '/^ *[^#]/{if(NF>=4&&$1=="*"&&$2=="*"){
-            printf "%s ", $4}}' "$DICT")
+if test -e "$DICT"; then
+    TEMPDICT=$(mktemp)
```
Comment (wdoekes) [inline ipmikvm]:
```
mktemp /dev/shm/ipmikvm.XXXXXX
```
That should write the temp file to memory only and not end up on a (unencrypted) drive.

**Review by wdoekes — CHANGES_REQUESTED** [review]
Looks good to go. Just the shm fix to avoid physical writing of passwords.

File: ipmikvm (hunk 1 above)
Comment (wdoekes) [reply ipmikvm]: Or, even better: https://github.com/ossobv/vcutil/pull/38#issuecomment-2129530922

Comment (quite) [reply ipmikvm]: Done.

Comment (quite) [reply ipmikvm]: > Or, even better: [#38 (comment)](https://github.com/ossobv/vcutil/pull/38#issuecomment-2129530922)

ah, will try

Comment (quite) [reply ipmikvm]: ok, seems workable!

File: ipmikvm [hunk 2]
```diff
@@ -126,21 +129,23 @@ IP=${1:-}; shift
...
-        test -z "$PASS" && PASS=$(echo "$LINES" | awk '{print $4}')
-    elif test -z "$USER" && test -z "$PASS"; then
-        # No user/pass supplied. Then get all the * * matches.
-        USERS=$(awk '/^ *[^#]/{if(NF>=4&&$1=="*"&&$2=="*"){
-            printf "%s ", $3}}' "$DICT")
-        PASSES=$(awk '/^ *[^#]/{if(NF>=4&&$1=="*"&&$2=="*"){
-            printf "%s ", $4}}' "$DICT")
+if test -e "$DICT"; then
+    DICTDATA=$(cat "$DICT")
+    if [ -n "$DICTDATA" ]; then
```
Comment (wdoekes) [inline ipmikvm]: `test`

File: ipmikvm [hunk 3]
```diff
@@ -126,21 +129,23 @@ IP=${1:-}; shift
...
+        # Alias match
+        LINES=$(printf '%s\n' "$DICTDATA" | awk "/^ *[^#]/{if(NF>=4&&(\$1==\"$IP\"||\$2==\"$IP\")){
```
Comment (wdoekes) [inline ipmikvm]: Mind the long lines please. max 79 columns.

File: ipmikvm [hunk 4]
```diff
@@ -126,21 +129,23 @@ IP=${1:-}; shift
...
+        if test -n "$LINES"; then
+            IP=$(echo "$LINES" | awk '{print $2}')
+            test -z "$USER" && USER=$(echo "$LINES" | awk '{print $3}')
+            test -z "$PASS" && PASS=$(echo "$LINES" | awk '{print $4}')
+        elif test -z "$USER" && test -z "$PASS"; then
+            # No user/pass supplied. Then get all the * * matches.
+            USERS=$(printf '%s\n' "$DICTDATA" | awk '/^ *[^#]/{if(NF>=4&&$1=="*"&&$2=="*"){
+                printf "%s ", $3}}')
+            PASSES=$(printf '%s\n' "$DICTDATA" | awk '/^ *[^#]/{if(NF>=4&&$1=="*"&&$2=="*"){
+                printf "%s ", $4}}')
```
Comment (wdoekes) [inline ipmikvm]: Here too.

**Review by wdoekes — CHANGES_REQUESTED** [review]

**Review by wdoekes — APPROVED** [review]

**Comment by wdoekes** [comment]
TY :+1:

**Pull request merged**